system/ee/EllisLab/Addons/comment/mod.comment.php

```diff
@@ -2251,23 +2258,38 @@ function comment_subscribe()
 		}
 
 		// They check out- let them through
-		ee()->subscription->$type();
+		if ($type == 'unsubscribe') {
+			ee()->subscription->$type(FALSE, $hash);
```

Comment:
According to a comment above, you must have the `$hash` if you are not logged in - which means you can be logged in and NOT have the hash, yet this line requires it - so do we need error checking around if a user is logged in but for some reason doesn't have a `$hash` in their URL (which would bypass the checks on 2215-2224)?

Reply:
I think it's ok- $hash can be false and if it's not set in get, it should be false.

In legacy/libraries/Subscription.php:

```	function unsubscribe($identifiers = FALSE, $hash = FALSE)
	{
		if ($hash != '')
		{
			ee()->db->where('hash', $hash);
		}
		else
		{
			$user = $this->_prep($identifiers);

```
So if $hash is false, it would be like just doing ee()->subscription->$type(); which was what the old code was doing.

Was I looking in the right spot re: the question?